src/whatsmyip/providers.py

```diff
@@ -30,6 +30,36 @@ def fetch():
         return ip
 
 
+class CloudflareDnsProvider(metaclass=IpProvider):
+    name = 'cloudflare-dns'
+
+    @staticmethod
+    def fetch():
+        r = dns.resolver.query('ns1.cloudflare.com')
+        ns_ip = r[0].address
+
+        resolver = dns.resolver.Resolver(configure=False)
+        resolver.nameservers = [ns_ip]
+
+        qr = resolver.query('whoami.cloudflare.com', 'TXT')
+
+        ip = qr.response.answer[0][0].to_text()
+        ip = ip.replace('"', '')
+        return ip
+
+
+class CloudflareHttpProvider(metaclass=IpProvider):
+    name = 'cloudflare'
```

Comment:
Thanks for your pull request! I really appreciate it!
Would you mind to make the name of the provider more explicit by changing the name to cloudflare-http?

```suggestion
    name = 'cloudflare-http'
```